### ISSUE TITLE
APItest/Twisted FutureWarnings

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#139]
**Reported by naughtont on 15 Sep 2005 19:30 UTC**
```
The APItest tools uses the TwistedMatrix libraries and
there is a warning generated from the current version
of Twisted shipped with oscar-4.2beta.  This is only a
warning and not an error.  I had planned on upgrading
APItest and using the newer Twisted RPMS that Will has
available, but not until after the oscar-4.2 release. 
I believe that should remove this warning msg by using
a newer version of Twisted.

Example Warning message:

/usr/lib/python2.3/site-packages/twisted/internet/defer.py:398:
FutureWarning: h
ex()/oct() of negative int will return a signed string
in Python 2.4 and up
  return "<%s at %s>" % (cname, hex(id(self)))

```

**Comment by dikim33** [comment]
**Comment by naughtont on 12 Oct 2005 16:45 UTC**
```

Note, a work around would be to call the python interpreter
directly, and add a "-W" command line option to ignore this
FutureWarning.

Example:
  cd packages/pvm/testing/
  python -W ignore::FutureWarning:: \
       /usr/local/apitest/apitest -T -f install_tests.apb 

This would be a workaround until we have the updated Twisted
libs, etc. (hopefully) for 5.0 but if not ready/willing then
this work around could be added to the testing driver
script(s) whereever apitest is being called.

Food for thought.

```

**Comment by dikim33** [comment]
**Comment by naughtont on 28 Oct 2005 20:10 UTC**
```

When I get time, I should upgrade APItest to 1.0.0 and use
the newer RPMS from sss-oscar-2.0 for Twisted, TwistedWeb,
ZopeInterfaces and python-elementtree (where necessary).

This will fix the future warnings and give us a *newer*
version of APItest.  
```

**Comment by dikim33** [comment]
**Comment by naughtont on 9 Jan 2006 17:14 UTC**
```

Updated to APItest v1.0 and supporting rpms, taken from
sss-oscar tree.    (Committed revision 4117.)

Needs to be tested in oscar trunk before closing this bug.

```

**Comment by dikim33** [comment]
**Comment by naughtont on 17 Jan 2006 15:58 UTC**
```

Since the fix for the latest APItest (to be
portable/standard) was to use the distro provided
'python-twisted' which is v1.3 based, the FutureWarnings may
occur on some systems.  

They are simply warning so I'm not going to stress about
them.  If needed I can apply the patch to Package.pm that
ignores these warning when running the apitest application
(python).
```

**Comment by dikim33** [comment]
**Comment by naughtont on 19 Jul 2006 21:02 UTC**
Are we still seeing FutureWarning? or have the distro provided 'python-twised' RPMS fixed this?

**Comment by dikim33** [comment]
**Comment by bli on 27 Jul 2006 00:20 UTC**
Still getting FutureWarning from RHEL4 Update 3:

```
Running Installation tests for pvm
/usr/lib/python2.3/site-packages/twisted/internet/defer.py:398: FutureWarning: hex()/oct() of negative int will return a signed string in Python 2.4 and up
  return "<%s at %s>" % (cname, hex(id(self)))
[      2006-07-26 17:11:43   pvmd-path-ls.apt
[PASS](PASS])       2006-07-26 17:11:44   envvar-pvm_arch.apt
[PASS]       2006-07-26 17:11:44   envvar-pvm_root.apt
```

**Comment by dikim33** [comment]
**Modified by jparpail on 6 Apr 2007 13:17 UTC**

**Comment by dikim33** [comment]
**Comment by jparpail on 10 Apr 2007 09:16 UTC**
Ooops,
Reverting to version "branch-4-2".

**Comment by dikim33** [comment]
**Modified by jparpail on 11 Apr 2007 08:25 UTC**

**Comment by dikim33** [comment]
**Comment by jparpail on 20 Jul 2007 13:51 UTC**
Thomas, I suggest that: either apply the patch to Package.pm to ignore the warning, either mark this bug as wontfix.

**Comment by dikim33** [comment]
**Modified by valleegr on 30 Jun 2008 06:35 UTC**

**Comment by dikim33** [comment]
**Modified by valleegr on 29 Sep 2009 07:14 UTC**
